include/gridtools/meta/fold.hpp

```diff
@@ -96,6 +96,36 @@ namespace gridtools {
             struct rfold<F, S, L<T1, T2, T3, T4, Ts...>> {
                 using type = F<T1, F<T2, F<T3, F<T4, typename rfold<F, S, L<Ts...>>::type>>>>;
             };
+#else
+            namespace fold_impl_ {
+                template <class>
+                struct id;
+                template <class>
+                struct state;
+                template <template <class...> class, class>
+                struct folder;
+                template <template <class...> class F, class S>
+                struct state<folder<F, S> &&> {
+                    using type = S;
+                };
+                template <template <class...> class F, class S, class T>
+                folder<F, F<S, T>> &&operator+(folder<F, S> &&, id<T> *);
+                template <template <class...> class F, class S, class T>
+                folder<F, F<T, S>> &&operator+(id<T> *, folder<F, S> &&);
+                template <class T>
+                T &&make();
+            } // namespace fold_impl_
+
+            template <template <class...> class F, class S, template <class...> class L, class... Ts>
+            struct lfold<F, S, L<Ts...>>
+                : fold_impl_::state<decltype(
+                      (fold_impl_::make<fold_impl_::folder<F, S>>() + ... + (fold_impl_::id<Ts> *)0))> {};
```

Comment:
What does the make do? Ensure lazy evaluation?

Reply:
wasn't really needed -- was just ad hoc `declval`. Fixed.